javav2/example_code/dynamodb/src/main/java/com/example/enhanced/dynamodb/QueryRecords.java

```diff
@@ -0,0 +1,179 @@
+
+public class QueryRecords {
+
+    private static final ProvisionedThroughput DEFAULT_PROVISIONED_THROUGHPUT =
+            ProvisionedThroughput.builder()
+                    .readCapacityUnits(50L)
+                    .writeCapacityUnits(50L)
+                    .build();
+
+    private static final TableSchema<Record> TABLE_SCHEMA =
+            StaticTableSchema.builder(Record.class)
+                    .newItemSupplier(Record::new)
+                    .addAttribute(String.class, a -> a.name("id")
+                            .getter(Record::getId)
+                            .setter(Record::setId)
+                            .tags(primaryPartitionKey()))
+                    .addAttribute(String.class, a -> a.name("sort")
+                            .getter(Record::getSort)
+                            .setter(Record::setSort)
+                            .tags(primarySortKey()))
+                    .addAttribute(String.class, a -> a.name("attribute")
+                            .getter(Record::getAttribute)
+                            .setter(Record::setAttribute))
+                    .addAttribute(String.class, a -> a.name("attribute2*")
+                            .getter(Record::getAttribute2)
+                            .setter(Record::setAttribute2)
+                            .tags(secondaryPartitionKey("gsi_1")))
+                    .addAttribute(String.class, a -> a.name("attribute3")
+                            .getter(Record::getAttribute3)
+                            .setter(Record::setAttribute3)
+                            .tags(secondarySortKey("gsi_1")))
+                    .build();
+
+    // QUery the Record table
+    public static void main(String[] args) {
+
+        //Create a DynamoDbClient object
+        Region region = Region.US_EAST_1;
+        DynamoDbClient ddb = DynamoDbClient.builder()
+                .region(region)
+                .build();
+
+        try{
+            // snippet-start:[dynamodb.java2.enhanced.query.main]
+            // Create a DynamoDbEnhancedClient and use the DynamoDbClient object
+            DynamoDbEnhancedClient enhancedClient = DynamoDbEnhancedClient.builder()
+                    .dynamoDbClient(ddb)
+                    .build();
+
+            //Create a DynamoDbTable object
+            DynamoDbTable<Record> mappedTable = enhancedClient.table("Record", TABLE_SCHEMA);
+
+            // Create a QueryConditional object that is used in the query operation
+            QueryConditional queryConditional = QueryConditional
+                    .keyEqualTo(Key.builder()
+                    .partitionValue("id-value")
+                    .build());
+
+            // Get items in the Record table and write out the ID value
+            Iterator<Page<Record>> results = mappedTable.query(QueryEnhancedRequest.builder()
+                    .queryConditional(queryConditional)
+                    .build()).iterator();
```

Comment:
This for this example I would use : 
```
Iterator<Record> results = mappedTable.query(queryConditional).items().iterator();
```

Then you don't have to worry about the pagination

Reply:
I rolled in this comment

Reply:
Updated code -- 

  // Get items in the Record table and write out the ID value
            Iterator<Record> results = mappedTable.query(queryConditional).items().iterator();

            while (results.hasNext()) {

                //Iterate through the Page object
                Record page = results.next();
                System.out.println("The record id is "+page.getId());
            }